tests/smcp_gold_panda.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"maistra/util"
+	"testing"
+	"time"
+
+	"istio.io/pkg/log"
+)
+
+func cleanuGoldPandaSMCP(namespace string) {
+	log.Info("# Cleanup ...")
+	util.KubeDelete(namespace, goldPandaSMCP, kubeconfig)
+	time.Sleep(time.Duration(waitTime*8) * time.Second)
+	// avoid namespace recreation for downstream service account settings
+}
+
+func TestGoldPandaSMCP(t *testing.T) {
+	defer cleanuGoldPandaSMCP("istio-gold-panda")
+
+	t.Run("Operator_test_smcp_gold_panda", func(t *testing.T) {
+
+		defer recoverPanic(t)
+
+		// // create a smcp with nil redudant policies field for Jaeger - See MAISTRA-1983
+		// util.CreateOCPNamespace("istio-gold-panda", kubeconfig)
+		// log.Info("Update SMCP with gold panda smcp")
+		// if err := util.KubeApply("istio-gold-panda", goldPandaSMCP, kubeconfig); err != nil {
+		// 	t.Errorf("Failed to deploy SMCP with goldpanda")
+		// 	log.Errorf("Failed to deploy SMCP with goldpanda")
+		// }
+		util.Shell(`kubectl patch -n %s smcp/%s --type merge -p '{"spec":{"techPreview":{"global":{"pathNormalization":"DECODE_AND_MERGE_SLASHES"}}}}'`, meshNamespace, smcpName)
+		log.Info("Update SMCP pathNormalization")
+
+	deployHttpbin("foo")
+	deploySleep("foo")
+
+	util.CheckPodRunning("foo", "app=httpbin", kubeconfig)
+	util.KubeApply("foo", goldPandaResource, kubeconfig)
+
+	log.Info("Verify setup")
+	for _, from := range namespaces {
+		for _, to := range namespaces {
+			sleepPod, err := util.GetPodName(from, "app=sleep", kubeconfig)
+			util.Inspect(err, "Failed to get sleep pod name", "", t)
+			cmd := fmt.Sprintf(`curl http://httpbin.%s:8000/ip -s -o /dev/null -w "sleep.%s to httpbin.%s: %%{http_code}"`,
```

Comment:
DECODE_AND_MERGE_SLASHES targets encoded characters, merging slashes, and merging `..`. This only addresses the one case. Should it address the others? 

Also, this addresses both allow and deny policies. Do we want to test both?

Reply:
yes, we can expend more path cases about the `slash and backslash characters (%2F, %2f, %5C and %5c)` 
For example , applying
 
```
apiVersion: security.istio.io/v1beta1
kind: AuthorizationPolicy
metadata:
name: deny-path-admin
spec:
action: DENY
rules:
- to:
    - operation:
        paths: ["/ip"]
```

Reply:
you don't need to create extra authorization policies. Just add curl commands with the special characters and confirm they pass pre-auth-policy and post-auth-policy. Same auth policy should work for everything